dask/templates/additional-worker-deployment.yaml

```diff
@@ -25,74 +25,81 @@ spec:
         app: {{ template "dask.name" $ }}
         release: {{ $.Release.Name | quote }}
         component: worker
-      {{- if $worker.annotations }}
+      {{- with $worker.annotations }}
       annotations:
-        {{- toYaml $worker.annotations | nindent 8 }}
+        {{- . | toYaml | nindent 8 }}
       {{- end }}
     spec:
+      {{- with $worker.image.pullSecrets }}
       imagePullSecrets:
-        {{- toYaml $worker.image.pullSecrets | nindent 8 }}
-      {{- if $worker.mounts.volumes }}
+        {{- . | toYaml | nindent 8 }}
+      {{- end }}
```

Comment:
This kind of change is also recurring in this PR: to not render fields with do-nothing like values. It is a valid k8s resource to declare for example that there is no imagePullSecrets to reference, but its also something you typically leave out if you don't have any such references.

Like this, the Helm charts generated k8s resources will be slimmer.